py-shared/pyproject.toml

```diff
@@ -0,0 +1,31 @@
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "quilt-shared"
+description = "Shared code for Quilt backend and lambdas"
+version = "0.1.0"
+requires-python = ">= 3.8, < 4"
+dependencies = [
+]
+
+[project.optional-dependencies]
+pydantic = [
+  "pydantic ~= 1.10",
+  "typing-extensions ~= 4.9",
+]
+boto = [
+  "boto3-stubs[s3] ~= 1.34",
+  "boto3 ~= 1.34",
+  "types-aiobotocore[s3] ~= 2.11",
```

Comment:
all of these seem to be needed only for type annotations
maybe move them to `dev`?

Reply:
well, yes, but that may complicate consumption of this package, since we'll need to list it in both the main section of the dependencies and then in some optional section with the `dev` extra. not sure how to make it convenient enough